APE26.rst

```diff
@@ -0,0 +1,325 @@
+
+We illustrate the new framework with the following pseudocode.
+
+.. code-block:: python
+
+    class BaseFrame:
+        ...
+
+    class ICRSFrame(BaseFrame):
+        pass  # no frame attributes
+
+    class FK5Frame(BaseFrame):
+        equinox: TimeAttribute
+
+    # ------
+
+    class BaseCoordinate:
+        frame: BaseFrame
+        data: BaseRepresentation
+
+    class Coordinate(BaseCoordinate):
+        ...  # it's fast.
+
+    class SkyCoord(BaseCoordinate):
+
+        def __init__(...):  # flexible input parsing
+            ...
+
+Branches and pull requests
+--------------------------
+No direct progress on these changes has yet occurred. Discussion
+of these ideas has however arisen in multiple issues and pull
+requests, demonstrating the need for and utility of the proposed
+changes.
+
+Several issues have been raised regarding topics such as
+confusion differentiating the use of ``frame`` and ``SkyCoord``
+for data storage, and problems arising in other astropy
+subpackages when using frames that store data. For example:
+
+- `Comparing Frame with data and SkyCoord with same data raises 
+  exception #13476 
+  <https://github.com/astropy/astropy/issues/13476>`_
+- `Add Frame objects without data to a Table #16823 
+  <https://github.com/astropy/astropy/issues/16823>`_
+
+Additionally, multiple pull requests have factored out common 
+code between frames and ``SkyCoord``, showing that there is no 
+proper separation of concern:
+
+- `Allow BaseCoordinateFrames to be stored in tables (by giving 
+  them .info) #16831 <https://github.com/astropy/astropy/pull/16831>`_
+- `Masked frames and SkyCoord #17106 
+  <https://github.com/astropy/astropy/pull/17016>`_ (this was later 
+  removed and instead methods were duplicated)
+
+Further, pull requests have added methods to make frames and 
+``SkyCoord`` even more similar, underscoring that frames *with* 
+data should not be separate entities from ``SkyCoord``:
+
+- `Implement BaseCoordinateFrame.to_table() #17009 
+  <https://github.com/astropy/astropy/pull/17009>`_
+- `Implement BaseCoordinateFrame.frame property #16356 
+  <https://github.com/astropy/astropy/pull/16356>`_
+
+In addition, many of these ideas have been developed and tested in
+parallal in the JAX-oriented library `coordinax
+<https://github.com/GalacticDynamics/coordinax>`_. Many of the
+developers of that library are also active Astropy developers and
+the development effort towards ``coordinax`` informs, tests, and
+validates the ideas presented in this APE. In short, it works.
+
+
+Implementation
+--------------
+The direct use of coordinate frames instead of ``SkyCoord`` is 
+common. In particular ``ICRS`` objects are frequently created 
+with data. Given the prevalent use, it is imperative to maintain 
+backward compatibility and not break the API too quickly. 
+Therefore, we propose implementing this APE through 4 steps (and 
+substeps).
+
+1. Splitting the frame classes into two hierarchies: ones with 
```

Comment:
For these steps, I would like to know more about what is your proposed timeline and exactly how break-y things will get at each step. And what would we do about helping to notify or patch affected downstream packages. So maybe even list each step as a sub-section if needed.

Even nicer is actual example usage code (not the implementation but *usage* of the implementation) and what the expected output would look like, for an example use case that is very common. Doesn't have to be fancy, just have to reflect what most of us use, which is usually something common like `SkyCoord(ra_arr, dec_arr, unit="deg")`.